cmd/pd-server/main.go

```diff
@@ -46,8 +49,53 @@ import (
 )
 
 func main() {
+	var ctx context.Context
+	var cancel context.CancelFunc
+	var svr basicsvr.Server
+
+	if len(os.Args) < 2 {
+		ctx, cancel, svr = createServerWrapper(os.Args[1:])
+	} else {
+		switch strings.ToLower(os.Args[1]) {
+		case "service-mode-tso":
+			ctx, cancel, svr = tsoserver.CreateServerWrapper(os.Args[2:])
```

Comment:
Why not just create another cmd `main.go` for TSO rather than reusing the `pd-server`?

Reply:
The current design is to use the same pd-server binary to support multi-service mode including tso, scheduling and admission control. This concern has been discussed in the design session, and the decision is made, as far as I know, for better compatibility and less migration cost especially for on-premises mode.

Reply:
Get it.